sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/AzureMonitorAspNetCoreEventSource.cs

```diff
@@ -74,5 +73,17 @@ public void GetEnvironmentVariableFailed(string envVarName, Exception ex)
 
         [Event(5, Message = "Failed to Read environment variable {0}, exception: {1}", Level = EventLevel.Error)]
         public void GetEnvironmentVariableFailed(string envVarName, string exceptionMessage) => WriteEvent(5, envVarName, exceptionMessage);
+
+        [Event(6, Message = "Failed to map unknown EventSource log level {0}", Level = EventLevel.Warning)]
+        public void MapLogLevelFailed(EventLevel level)
+        {
+            if (IsEnabled(EventLevel.Warning))
+            {
+                WriteEvent(6, level.ToString());
+            }
+        }
```

Comment:
might be a problem with this. On line 82 you're passing a string to WriteEvent. But the Event definition on line 77 has a parameter of EventLevel. 

1. I think this will fail because the types are different.
2. I don't think you can use an `EventLevel` in the params because it's not a primitive type. 

Here's an example of how I would solve this: https://github.com/Azure/azure-sdk-for-net/blob/d07d15dfbae911a84f4f36a27331066184f1b941/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.cs#L62-L72

Reply:
Yay, our unit tests caught this :)

```
Azure.Monitor.OpenTelemetry.AspNetCore.Tests.AzureMonitorAspNetCoreEventSourceTests.EventSourceTest_AzureMonitorExporterEventSource [FAIL]
System.Exception : Method 'AzureMonitorAspNetCoreEventSource.MapLogLevelFailed' is implemented incorrectly.
```